libhb/qsv_common.c

```diff
@@ -645,6 +646,54 @@ static int query_capabilities(mfxSession session, mfxVersion version, hb_qsv_inf
                         "hb_qsv_info_init: mfxExtCodingOption2 check failed (0x%"PRIX32", 0x%"PRIX32", %d)\n",
                         info->codec_id, info->implementation, status);
             }
+
+            /*
+             * Determine whether mfxExtCodingOption2 and its fields for lookahead are supported
+             */
+            if (HB_CHECK_MFX_VERSION(version, 1, 6)) // && info->capabilities & HB_QSV_CAP_RATECONTROL_LA)
+            {
+                init_video_param(&inputParam);
+                inputParam.mfx.CodecId = info->codec_id;
+                inputParam.mfx.RateControlMethod = MFX_RATECONTROL_LA;
+
+                memset(&extCodingOption2, 0, sizeof(extCodingOption2));
+                extCodingOption2.Header.BufferId = MFX_EXTBUFF_CODING_OPTION2;
+                extCodingOption2.Header.BufferSz = sizeof(mfxExtCodingOption2);
+                extCodingOption2.LookAheadDS = MFX_LOOKAHEAD_DS_4x;
+
+                inputParam.ExtParam    = videoExtParam;
+                inputParam.ExtParam[0] = (mfxExtBuffer*)&extCodingOption2;
+                inputParam.NumExtParam = 1;
+
+                memset(&videoParam, 0, sizeof(mfxVideoParam));
+                videoParam.mfx.CodecId = inputParam.mfx.CodecId;
+                memset(&extCodingOption2Out, 0, sizeof(extCodingOption2Out));
+                extCodingOption2Out.Header.BufferId = MFX_EXTBUFF_CODING_OPTION2;
+                extCodingOption2Out.Header.BufferSz = sizeof(mfxExtCodingOption2);
+
+                videoParam.ExtParam    = videoExtParamOut;
+                videoParam.ExtParam[0] = (mfxExtBuffer*)&extCodingOption2Out;
+                videoParam.NumExtParam = 1;
+
+                status = MFXVideoENCODE_Query(session, &inputParam, &videoParam);
```

Comment:
it seems that this call is not always successful, 

@DOram424 no "check failed" in your tests?
@agalin89 can you pls suggest changes for latest drivers while on Windows?

Reply:
yes, it failed with the following message:
hb_qsv_info_init: mfxExtCodingOption2 check failed (0x43564548, 0x104, -3) for lookahead rate control

LA is available in two ways:
•	As a part of AVC encoder
•	As a standalone pre-analysis plug-in

Please follow https://github.com/Intel-Media-SDK/MediaSDK/issues/1743 
Shortly to use LA for HEVC codec you need to have two sessions and load plugin with ID MFX_PLUGINID_H264LA_HW in a separate session. Please close the sessions after query calls.

Reply:
The check failed stuff should be fixed now if you'd like to try again. I can be a bit of a space cadet, so please do.

As to the lookahead stuff. I'm not 100% sure I understand your concern here. Correct me if I'm wrong/misunderstanding but currently handbrake only supports lookahead for x264 encodes and the plugin version is .. just not supported.

It's out of scope for this patch to change the behaviour of handbrake wrt to lookahead but 
I may have a future patch where it becomes in scope so am very interested to hear if you found any noticeable improvements from lookahead with hevc B pyramid encodings and long gops (128+). Given the bi-directional inter-frame matching I'd expect a lot less benefit compared to x264.